lib/HTFeed/Stage/ImageRemediate.pm

```diff
@@ -369,7 +376,152 @@ sub _remediate_tiff {
       && $self->repair_tiff_exiftool( $infile, $outfile, $self->{newFields} );
 
     return $ret;
+}
+
+# Extract metadata from jpg/png and write to the corresponding tiff.
+sub copy_metadata {
+    my $self    = shift;
+    my $ext     = shift;
+    my $infile  = shift;
+    my $outfile = shift;
 
+    $self->{oldFields} = $self->get_exiftool_fields($infile);
+    $self->{newFields} = {};
+
+    # Delegate to the method that knows how to extract from a ".$ext"
+    if ($ext eq ".jpg") {
+	$self->{newFields} = extract_jpg_metadata($self->{oldFields});
+    } elsif ($ext eq ".png") {
+	$self->{newFields} = extract_png_metadata($self->{oldFields});
+    } else {
+	croak "copy_metadata knows not extension: $ext";
+	return;
+    }
+
+    # Write extracted metadata to outfile.
+    my $exifTool = new Image::ExifTool;
+    while (my ($field, $val) = each(%{$self->{newFields}})) {
+	print "($ext) setting exif field $field to $val\n";
+	my ($success, $errStr) = $exifTool->SetNewValue($field, $val);
+	if (defined $errStr) {
+	    croak("Error setting new tag $field => $val: $errStr\n");
+	    return 0;
+	}
+    }
+    my $exif_write_status = $exifTool->WriteInfo($outfile);
+    unless ($exif_write_status == 1) {
+	get_logger()->trace("Failed EXIF write to $outfile");
+    }
+}
+
+# Build and return a hash of jpg metadata.
+sub extract_jpg_metadata {
+    my $olf = shift; # ref to $self->{oldFields}, a hash of exiftool data
```

Comment:
Here I mostly just don't know enough about image metadata to figure it out.

Reply:
I made the extract_x_metadata methods a little tighter, so now they only get the resolution-related metadata:
```
'IFD0:ResolutionUnit'    
'IFD0:XResolution'       
'IFD0:YResolution'       
'XMP-tiff:ResolutionUnit'
'XMP-tiff:XResolution'   
'XMP-tiff:YResolution'   
```